src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -73,10 +79,62 @@ SimpleDataflowCPU::activateContext(ThreadID tid)
     BaseCPU::activateContext(tid);
 }
 
+void
+SimpleDataflowCPU::attemptAllDataXlations()
+{
+    DPRINTF(SDCPUCoreEvent, "attemptAllDataXlations()\n");
+    assert(!dataTranslationReqs.empty());
+
+    while (!dataTranslationReqs.empty()) {
+        auto xlation_req = dataTranslationReqs.front();
+        RequestPtr req = xlation_req.request;
+        ThreadContext* tc = xlation_req.tc;
+        bool write = xlation_req.write;
+        TranslationCallback callback = xlation_req.callback;
+
+        DPRINTF(SDCPUCoreEvent, "handleDataAddrTranslation()\n");
+
+        // Retrieve dtb from the ThreadContext
+        BaseTLB* dtb = tc->getDTBPtr();
+
+        // The event callback for when the translation completed comes through
+        // a BaseTLB::Translation object, so we pass that callback straight to
+        // the requestor's callback function by subclassing it.
+        BaseTLB::Translation* handler = new CallbackTransHandler(callback,
+                                                                 true);
+
+        dtb->translateTiming(req, tc, handler,
+                             write ? BaseTLB::Write : BaseTLB::Read);
+
+        dataTranslationReqs.pop_front();
+    }
+}
+
+void
+SimpleDataflowCPU::attemptAllExecutions()
+{
+    DPRINTF(SDCPUCoreEvent, "attemptAllExecutions()\n");
+    assert(!executionReqs.empty());
+
+    while (!executionReqs.empty()) {
+        auto req = executionReqs.front();
+
+        Event* event = new EventFunctionWrapper([this, req]() {
+            completeExecution(req);
+        }, name() + ".executeEvent", true);
+
+        schedule(event, clockEdge(executionLatency));
```

Comment:
This should respect the parameter clocked_execution. If it's not clocked_execution, it should probably be curTick() + clockPeriod() * executionLatency or something like that.

Reply:
or curTick() + cyclesToTicks(executionLatency)

Reply:
they're actually the same, just one is in-lined I guess